src/main/java/com/puppycrawl/tools/checkstyle/utils/InlineTagUtils.java

```diff
@@ -0,0 +1,150 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.api.LineColumn;
+
+/**
+ * Tools for extracting inline tags from Javadoc comments.
+ *
+ * @author Nathan Naze
+ */
+final class InlineTagUtils {
+
+    /**
+     * Inline tag pattern.
+     */
+    private static final Pattern INLINE_TAG_PATTERN = Pattern.compile(
+            ".*?\\{@(\\p{Alpha}+)\\b(.*?)}", Pattern.DOTALL);
+
+    /** Pattern to recognize leading "*" characters in Javadoc. */
+    private static final Pattern JAVADOC_PREFIX_PATTERN = Pattern.compile(
+        "^\\s*\\*", Pattern.MULTILINE);
+
+    /** Pattern matching whitespace, used by {@link InlineTagUtils#collapseWhitespace(String)}. */
+    private static final Pattern WHITESPACE_PATTERN = Pattern.compile("\\s+");
+
+    /** Pattern matching a newline. */
+    private static final Pattern NEWLINE_PATTERN = Pattern.compile("\\n");
+
+    /** Line feed character. */
+    private static final String LINE_FEED = "\n";
+
+    /** Carriage return character. */
+    private static final String CARRIAGE_RETURN = "\r";
+
+    /** Non instantiable. */
+    private InlineTagUtils() { }
+
+    /**
+     * Extract inline Javadoc tags from the given comment.
+     * @param lines The Javadoc comment (as lines).
+     * @return The extracted inline Javadoc tags.
+     */
+    public static List<TagInfo> extractInlineTags(String... lines) {
```

Comment:
All other methods should be private.
Coverage should be archived by test cases for `extractInlineTags`